src/components/ConfirmPopover/ConfirmPopover.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Popover } from 'react-tiny-popover'
+import { Button } from '../Button'
+import ErrorBoundary from '../../containers/ErrorHOC/ErrorHOC'
+
+import './ConfirmPopover.scss'
+import { Icon } from '../Icon'
+
+export default class ConfirmPopover extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isOpen: false,
+    }
+  }
+
+  static propTypes = {
+    title: PropTypes.string,
+    text: PropTypes.string,
+    backText: PropTypes.string,
+    confirmText: PropTypes.string,
+    danger: PropTypes.bool,
+    padding: PropTypes.number,
+    positions: PropTypes.arrayOf(PropTypes.string),
+    align: PropTypes.string,
+    icon: PropTypes.string,
+    onClose: PropTypes.func,
+    onConfirm: PropTypes.func,
+  }
+
+  static defaultProps = {
+    title: 'Are you sure you want to proceed?',
+    text: '',
+    backText: 'Go back',
+    confirmText: 'Continue',
+    danger: false,
+    padding: undefined,
+    positions: undefined,
+    align: undefined,
+    icon: undefined,
+    onClose: () => {},
+    onConfirm: () => {},
+  }
+
+  close = () => {
+    this.setState({ isOpen: false })
+  }
+
+  onConfirmClick = async () => {
+    this.setState({ loading: true })
+    await this.props.onConfirm()
+    this.setState({ isOpen: false, loading: false })
```

Comment:
yep, it will close the popover after onConfirm finishes. Now that you ask though, I realized that if onConfirm fails, it isn't catching that. I'll add a try catch block